dotcms-integration/src/test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java

```diff
@@ -1395,4 +1394,39 @@ public void findAllFromDB() throws DotDataException, DotSecurityException {
                 siteListWithSystemHost.size() - siteList.size());
     }
 
+    /**
+     * <ul>
+     *     <li><b>Method to test: </b>{@link HostAPI#findByIdOrKey(String, User, boolean)}</li>
+     *     <li><b>Given Scenario: </b>Find the Default Site using either its Identifier or its
+     *     Key.</li>
+     *     <li><b>Expected Result: </b>The Site API must be able to find the specified Site using
+     *     either the Identifier or the Site Key.</li>
+     * </ul>
+     */
+    @Test
+    public void findSiteByIdOrKey() throws DotDataException, DotSecurityException {
+        // ╔══════════════════╗
+        // ║  Initialization  ║
+        // ╚══════════════════╝
+        final HostAPI siteAPI = APILocator.getHostAPI();
+        final User systemUser = APILocator.systemUser();
+        final Host defaultSite = siteAPI.findDefaultHost(systemUser, false);
+        final String defaultSiteId = defaultSite.getIdentifier();
+        final String defaultSiteKey = defaultSite.getHostname();
+
+        // ╔════════════════════════╗
+        // ║  Generating Test data  ║
+        // ╚════════════════════════╝
+        final Optional<Host> siteById = siteAPI.findByIdOrKey(defaultSiteKey, systemUser, false);
+        final Optional<Host> siteByKey = siteAPI.findByIdOrKey(defaultSiteId, systemUser, false);
+
+        // ╔══════════════╗
+        // ║  Assertions  ║
+        // ╚══════════════╝
+        assertTrue("The Site by ID was NOT found!", siteById.isPresent());
+        assertTrue("The Site by Key was NOT found!", siteByKey.isPresent());
+        assertEquals("Site by ID and Site by Key must point to the Default Site",
```

Comment:
Let's also compare between both `siteById.get().getIdentifier()` and `siteByKey.get().getIdentifier()` against `defaultSiteId` to make sure we are getting the correct site from the AAPI